js/script.js

```diff
@@ -0,0 +1,176 @@
+		return player;
+	}
+}
+
+
+var turnCheck = function(){   // this function changes the position of the turn indicator 
+	if(isPlayerOne){
+		document.getElementById("x-span").className = 'active-turn';
+		document.getElementById("o-span").className = '';
+	} else {
+		document.getElementById("x-span").className = '';
+		document.getElementById("o-span").className = 'active-turn'
+	}
+}
+
+var changeBox = function(cell) {     // this assigns a square to x or o
+	if (!gameOver && cell.className.indexOf("cell") === -1) {  // this line prevents overwriting played squares
+		if (isPlayerOne){
+			cell.className += " x-cell";  //gives the square P1 color
+			//cell.innerHTML = "<p>X<p>";   //gives the square an X
+			isPlayerOne = false;          //makes it P2's turn
+			checkForWinner();
+			turns++;
+			computerMove();				//prompts computer to take its turn (sigle-player only!)
+		} else {
+			cell.className += " o-cell";
+			//cell.innerHTML = "<p>O<p>";
+			isPlayerOne = true;	
+			turns++;
+		}
+	}
+	
+	turnCheck();
+}
+
+
+var clearBoard = function(){  //function for the clear button
+	for(var i in allBoxes){
+		allBoxes[i].className = 'play-square'; //clears all x && o-cell classes from boxes
+		allBoxes[i].innerHTML = ''; //clears all text from boxes
+		isPlayerOne = true;         //makes it player one's turn
+		turnCheck();
+		turns = 0;
+		gameOver = false;
+		document.getElementById('winner-name').innerHTML = "Who will survive?";
+	}
+}
+
+var modeTogle = function(){
+	if(!singlePlayerMode){
+		document.getElementById('single-player').innerHTML = "2p Mode";
+		singlePlayerMode = true;
+		if(!isPlayerOne){
+			computerMove();
+			turnCheck();
+		}
+	} else if (singlePlayerMode){
+		document.getElementById('single-player').innerHTML = "Vs. Comp";
+		singlePlayerMode = false;
+	}
+}
+
+
+
+// call functions
+
+turnCheck();
+
+// event listeners
+
+document.getElementById('clear-button').addEventListener('click', function(){
+	clearBoard();
+})
+
+document.getElementById('single-player').addEventListener('click', function(){
+	modeTogle();
+})
+
+document.getElementById('cell-1').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-2').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-3').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-4').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-5').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-6').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-7').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-8').addEventListener('click', function(){
+	changeBox(this);
+})
+
+document.getElementById('cell-9').addEventListener('click', function(){
+	changeBox(this);
+})
```

Comment:
Another method that would work would be to get the elements by class name, then iterate over the array using a loop. Example:

```js
var allBoxes = document.getElementsByClassName('play-square');

for (var i = 0; i < allBoxes.length; i++) {
  allBoxes[i].addEventListener('click', function() {
    //do stuff
  }
}
```

We'll also learn this week about event delegation in the context of jQuery, which only requires one event handler.